### PR TITLE
Vertical Ventilation - removing plating using a fireaxe

### PR DESCRIPTION
**Pull request by west3436**
## What this does

Wielded fire axes can now remove plating, exposing the space tile beneath it. The process takes 6seconds to complete, or 3 if you are strong (RCD plating deconstruction takes 5 seconds, for reference).

## Why it's good

Now that fires are more prevalent thanks to the autoignition enhancements, it has been difficult to cool a room which has had a fire as scrubbers and air conditioners act very slowly. IRL, firefighters use ventilation to remove hot gasses from a room. Horizontal ventilation is already possible in game using the fire axe on a window and grille; but if the room does not have any outside-facing windows, then vertical ventilation through the floor is required. This change allows atmos techs to vent the gasses out of the room quickly by removing the floor without having to carry an RCD, at the cost of slightly more time unless you are fit. The fire axe currently has limited use in firefighting, so this addition gives it more versatility.

## Changelog

:cl:
 * rscadd: Fire axes can now remove floor plating.

### CODE REVIEW

**Comment by Eneocho** [comment]
While I can see the reasoning behind the axe breaking, I don't like it. Axes are a rare resource, so having it break after 1 or 2 forceful breaches just feels outta place.

**Comment by west3436** [comment]
> While I can see the reasoning behind the axe breaking, I don't like it. Axes are a rare resource, so having it break after 1 or 2 forceful breaches just feels outta place.

You can order axes at cargo for like 15 credits.

**Comment by Eneocho** [comment]
And you can order eguns for like 40 credits, still not a reason to make them weak as hell.

**Comment by CrazyAmphibian** [comment]
i like the concept, and the fireaxe doesn't actually have that much utility for firefighting, but i feel like just allowing for deconstruction of tiles would be the better play (maybe with a more exclusive engineering item like a socket wrench to reduce shittery?), it never made too much sense as to why you can't turn them back into sheets like you can with walls and other machines.
though a 50% chance to break is just annoying. I think you could just have it take a while to break like what was done with haligans and the shutters.

**Comment by Kurfursten** [comment]
Given their unwieldy size and limited quantity, I think it would be fine to just let fireaxes remove plating with no risk and at the same speed as an RCD. Yes you can get fireaxes from cargo... you can also print an RCD at cargo which also removes floors. This DOES make the fireaxe the best at removing platings... but it's also just a niche. RCDs can remove a floor without bothering with the plating and build things as well.

The only criticism of this is that someone could sit around slowly deconstructing all the floors, but quite frankly that kind of behavior doesn't happen much. You COULD weld or drill every wall on z1 but people aren't doing it.

**Comment by DorainGrey** [comment]
Remove the breaking I think.

Spessmen need a way to place glass floors without an RCD.

**Comment by west3436** [comment]
Alright, sounds like pretty unanimous feedback to remove the break chance. Not sure if I'll have time to get to it until Sunday but I'll remove the chance of breaking and bring the removal time down closer to the RCD.

**Comment by Dacendeth** [comment]
I think changing it from 20/10 if strong to 10/5 if strong is fine.

**Comment by Kurfursten** [comment]
I'd go as far as 6 / 3, which is aggressively statted, but I have reasons to believe this is fine. 5 is the RCD baseline and 4 is the time it takes to deconstruct girders. There should be no objection to the idea that deconstructing floors 50% slower than you can girders is that dangerous given that this has to be wielded. Strong gene is hard to get, too, so it should be well rewarded.

Balance this by giving it a unique sound and animation so people can see it's happening.

**Comment by Eneocho** [comment]
requesting that if you go hulk you can just HULK SMASH (with precision) and break through the plating the fastest

**Comment by west3436** [comment]
Removed the axe breaking chance. Updated OP.

**Comment by SonixApache** [comment]
i see this only being used for evil, but who knows

**Comment by CrazyAmphibian** [comment]
> i see this only being used for evil, but who knows

i definitely see the potential for abuse, but i think that this feature will go slept upon because of how niche it is.
fire axes are less accessible than RCDs (cargo order with access requirement versus hacked autolathe),
fire axes are more inconspicuous than an RCD
fire axes are more unwieldy than an RCD (cannot be placed in a bag)

and as for an atmos tech actually firefighting, the RCD is easier since you won't have to juggle an axe around on the floor in your hand slot (and takes less time), and atmos techs and engineers basically being the same job, nobody will find it \*as\* weird that you're running around the station with an RCD versus an axe.

i think that, to give this a bit more of a reason to be used instead of the RCD, that you should also drop a metal tile so that this would be a more resource efficient route (matter cartridges use a lot of (cheap) resources to make) at the cost of time and tools.

File: code/game/objects/items/weapons/twohanded.dm [hunk 1]
```diff
@@ -106,6 +106,21 @@
...
+		var/turf/simulated/floor/T = A
+		to_chat(viewers(user), "<span class='danger'>[user] begins to remove the plating using \the [src]!</span>")
+		var/breaktime = 6 SECONDS
+		if(istype(user,/mob/living/carbon/human))
+			var/mob/living/carbon/human/H = user
+			if(H.get_strength() >= 2)
+				breaktime = 3 SECONDS
+		if(!do_after(user, T, breaktime, 3, custom_checks = new /callback(src, /obj/item/weapon/fireaxe/proc/on_do_after)))
+			return
+		playsound(src, 'sound/items/Deconstruct.ogg', 50, 1)
```
Comment (Kurfursten) [inline code/game/objects/items/weapons/twohanded.dm]: Pick a better sound and I will merge this

Comment (west3436) [reply code/game/objects/items/weapons/twohanded.dm]: Sure, I'll dig something up

Comment (west3436) [reply code/game/objects/items/weapons/twohanded.dm]: New and improved ogg

**Pull request merged**